safe_transaction_service/history/management/commands/setup_service.py

```diff
@@ -5,37 +5,67 @@
 from django.core.management.base import BaseCommand
 from django.db.models import Min
 
-from django_celery_beat.models import IntervalSchedule, PeriodicTask
+from django_celery_beat.models import CrontabSchedule, IntervalSchedule, PeriodicTask
 
 from gnosis.eth import EthereumClientProvider
 from gnosis.safe.addresses import MASTER_COPIES, PROXY_FACTORIES
 
 from ...models import IndexingStatus, ProxyFactory, SafeMasterCopy
 
 
+@dataclass
+class CronDefinition:
+    minute: str = "*"
+    hour: str = "*"
+    day_of_week: str = "*"
+    day_of_month: str = "*"
+    month_of_year: str = "*"
+
+
 @dataclass
 class CeleryTaskConfiguration:
     name: str
     description: str
-    interval: int
-    period: str
+    interval: int = 0
+    period: str = None
+    cron: CronDefinition = None
     enabled: bool = True
 
     def create_task(self) -> Tuple[PeriodicTask, bool]:
-        interval_schedule, _ = IntervalSchedule.objects.get_or_create(
-            every=self.interval, period=self.period
-        )
-        periodic_task, created = PeriodicTask.objects.get_or_create(
-            task=self.name,
-            defaults={
+        if self.cron is None:
```

Comment:
I think it's better to check what you are adding and not that the other attribute is not defined
```suggestion
        assert self.period or self.cron, "Task must define period or cron"
        if self.period:
```